stream_in.c

```diff
@@ -397,12 +397,13 @@ int in_create(struct audio_hw_device *dev,
     if ((xin->p_handle == NULL) || (!pcm_is_ready(xin->p_handle))) {
         ALOGE("%s() failed (%d). Can't open stream on device.", __FUNCTION__, errno);
         if (xin->p_handle != NULL) {
+            ALOGE("pcm error:'%s'", pcm_get_error(xin->p_handle));
             pcm_close(xin->p_handle);
             xin->p_handle = NULL;
         }
         *stream_in = NULL;
         free(xin);
-        return -ENOMEM;
+        return -errno;
```

Comment:
Consider the possibility of changing errno by free()

Reply:
Thanks. Fixed.